packages/ra-ui-materialui/src/field/BooleanField.tsx

```diff
@@ -93,8 +93,8 @@ export interface BooleanFieldProps
         Omit<TypographyProps, 'textAlign'> {
     valueLabelTrue?: string;
     valueLabelFalse?: string;
-    TrueIcon?: SvgIconComponent | null;
-    FalseIcon?: SvgIconComponent | null;
+    TrueIcon?: SvgIconComponent | boolean;
+    FalseIcon?: SvgIconComponent | boolean;
```

Comment:
I think it would be better to only support `SvgIconComponent | false`, because it wouldn't make much sense to set one of these props to `true`, would it?